dotnet/src/VectorDataIntegrationTests/SqliteVecIntegrationTests/Support/SqliteTestEnvironment.cs

```diff
@@ -1,50 +1,34 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 
-using System.Data;
 using Microsoft.Data.Sqlite;
 
 namespace SqliteVecIntegrationTests.Support;
 
 internal static class SqliteTestEnvironment
 {
-    private static bool? s_isSqliteVecInstalled;
+    private static readonly Lazy<bool> s_canUseSqlite = new(CanCreateConnectionAndLoadExtension);
 
-    internal static bool TryLoadSqliteVec(SqliteConnection connection)
+    internal static bool CanUseSqlite => s_canUseSqlite.Value;
+
+    private static bool CanCreateConnectionAndLoadExtension()
     {
-        if (!s_isSqliteVecInstalled.HasValue)
+        try
         {
-            if (connection.State != ConnectionState.Open)
-            {
-                throw new ArgumentException("Connection must be open");
-            }
-
-            try
-            {
-                connection.LoadVector();
-                s_isSqliteVecInstalled = true;
-            }
-            catch (SqliteException)
-            {
-                s_isSqliteVecInstalled = false;
-            }
+            using var connection = new SqliteConnection("Data Source=:memory:;");
+            connection.Open();
+            connection.LoadVector();
         }
-
-        return s_isSqliteVecInstalled.Value;
-    }
-
-    internal static bool IsSqliteVecInstalled
-    {
-        get
+        catch (TypeInitializationException ex)
```

Comment:
So far we were assuming that `SqliteConnection` can be used an maybe the load of the extension will just fail.

It turned out that on Full Framework we just may not be able to use `SqliteConnection` itself due to some missing native dependencies in the NuGet package